lib/server/swarm.js

```diff
@@ -7,12 +7,35 @@ var randomIterate = require('random-iterate')
 // Regard this as the default implementation of an interface that you
 // need to support when overriding Server.createSwarm() and Server.getSwarm()
 function Swarm (infoHash, server) {
+  this.infoHash = infoHash
+  this.complete = 0
+  this.incomplete = 0
+
   this.peers = new LRU({
     max: server.peersCacheLength || 1000,
-    maxAge: server.peersCacheTtl || 900000 // 900 000ms = 15 minutes
+    maxAge: server.peersCacheTtl || 20 * 60 * 1000 // 20 minutes
+  })
+
+  // When a peer is evicted from the LRU store, send a synthetic 'stopped' event
+  // so the stats get updated correctly.
+  this.peers.on('evict', function (data) {
+    var peer = data.value
+    this.announce({
```

Comment:
This won't work, peer is already evicted. _onAnnounce gets it and passes it to _onAnnounceStopped as null/undefined.

We should only perform what's inside _onAnnounceStopped:

```js
if (peer.complete) this.complete -= 1
else this.incomplete -= 1
```